gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1531,6 +1545,21 @@ private Credential getCredential(Configuration config)
     return Optional.of(impersonatedCredential.createScoped(CredentialFactory.GCS_SCOPES));
   }
 
+  /**
+   * @param name login user name or group name
+   * @return true if name matches with currently login user or any group.
+   */
+  private static boolean matchedUserOrGroup(String name) throws IOException {
+    String userName = UserGroupInformation.getCurrentUser().getShortUserName();
+    String[] groupNames = UserGroupInformation.getCurrentUser().getGroupNames();
```

Comment:
Aren't this a little bit risky if we allow such ambiguity in security configuration? Should we clearly distinguish between the two? Also, may you check how S3 is configuring this, so we will have as close user-facing configuration to them as possible (if it make sense)?

Reply:
Resolved. Having separate property for each solved this problem